src/main/scala/com/guidewire/cda/config/ClientConfigReader.scala

```diff
@@ -75,14 +78,18 @@ object ClientConfigReader {
    * @return Config case class with config fields
    */
   private[cda] def parseConfig(configPath: String): ClientConfig = {
-    val configSource = Source.fromFile(configPath)
     try {
-      val configYaml = configSource.getLines.mkString("\n")
+      val uri = new URI(configPath)
+      val configYaml = uri.getScheme match {
+        case "s3" => AWSUtils.S3Utils.getFileAsString(configPath)
+        case _    =>
+          val configSource = Source.fromFile(configPath)
+          configSource.getLines.mkString("\n")
+      }
       ObjectMapperSupplier.yamlMapper.readValue(configYaml, classOf[ClientConfig])
     } catch {
       case e: InvalidFormatException => throw InvalidConfigParameterException(s"There was an error while parsing the config file ($configPath), look at the CausedBy exception for details.", e)
-    } finally {
-      configSource.close
+      case i: Exception              => throw i
```

Comment:
```suggestion
      case ex: Exception              => throw ex
```
Exception does have an `i`, but doesn't seem intuitive to me.